scripts/run_gw.py

```diff
@@ -42,20 +43,34 @@ def log_to_pt(message):
     with open(scriptvars_json_file) as json_file:
         scriptvars_json = json.load(json_file)
 else:
-    print("INFO: '%s' does not exist." % scriptvars_json_file)
-
-print("%s: command to run is: '%s'" % (script_name, " ".join(cmd_arr)))
-
-# run command
-rc = None
-proc = subprocess.Popen(cmd_arr,
-                        env=scriptvars_json,
-                        stdout=subprocess.PIPE,
-                        stderr=subprocess.STDOUT,
-                        )
-while rc == None:
-    line = proc.stdout.readline()
-    log_to_pt(line.rstrip())
-    rc = proc.poll()
-sys.stdout.flush()
-sys.exit(rc)
+    print("%s/INFO: '%s' does not exist." % (script_name, scriptvars_json_file))
+
+# continue until we run a non-superseded task
+while True:
+    supersede_seen = False
+    rc = None
+
+    print("%s/INFO: command to run is: '%s'" % (script_name, " ".join(cmd_arr)))
+    # run command
+    proc = subprocess.Popen(cmd_arr,
+                            env=scriptvars_json,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.STDOUT,
+                            )
+
+    while rc == None:
+        line = proc.stdout.readline()
+        stripped_line = line.rstrip()
+        log_to_pt(stripped_line)
+        if 'has been superseded' in stripped_line.lower():
+            supersede_seen = True
+        rc = proc.poll()
+    sys.stdout.flush()
+    # exit if the rc is non-zero (even if superseded)
+    if rc != 0:
+        sys.exit(rc)
+    # if we've processed a real job, exit out of while True
+    if not supersede_seen:
+        sys.exit(rc)
+    else:
+        print("%s/INFO: task was superseded, running again..." % script_name)
```

Comment:
I'd prefer something like
```
    # exit if the rc is non-zero (even if superseded) or if we've processed a real job
    # otherwise consume another task.
    if rc != 0 or not supersede_seen:
        break
    print("%s/INFO: task was superseded, running again..." % script_name)
 
sys.exit(rc)
```

Let's also change the variable name from supersede_seen to just superseded.

Reply:
Done.